client.go

```diff
@@ -65,9 +68,19 @@ func (e SecretDeleted) Error() string {
 	return "deleted"
 }
 
+func (c Client) failCountInc() {
+	metrics.GetOrRegisterCounter("runtime.server.fails", c.metricsHandle.Registry).Inc(1)
+}
+func (c Client) failCountClear() {
+	metrics.GetOrRegisterCounter("runtime.server.fails", c.metricsHandle.Registry).Clear()
```

Comment:
Instead of calling this over and over again, just call it once and then reuse it?